src/librustc_parse/parser/diagnostics.rs

```diff
@@ -331,6 +331,7 @@ impl<'a> Parser<'a> {
                         Applicability::MachineApplicable
                     },
                 );
+                self.sess.type_ascription_path_suggestions.borrow_mut().insert(sp);
```

Comment:
What happens if the `err` is cancelled and never emitted?
Will we get an ICE from `delay_as_bug` in the worst case?

Reply:
Correct.